src/CarbonAware.CLI/src/Commands/Emissions/EmissionsCommand.cs

```diff
@@ -41,17 +71,60 @@ private async Task Run(InvocationContext context)
         var locations = context.ParseResult.GetValueForOption<string[]>(_requiredLocation);
         var startTime = context.ParseResult.GetValueForOption<DateTimeOffset?>(_startTime);
         var endTime = context.ParseResult.GetValueForOption<DateTimeOffset?>(_endTime);
-        
-        var parameters = new CarbonAwareParametersBaseDTO() { 
-            MultipleLocations = locations,
+        var best = context.ParseResult.GetValueForOption<bool>(_best);
+        var average = context.ParseResult.GetValueForOption<bool>(_average);
+        var parameters = new CarbonAwareParametersBaseDTO()
+        {
             Start = startTime,
             End = endTime
         };
-
         // Call the aggregator.
-        var results = await aggregator.GetEmissionsDataAsync(parameters);
+        if (best)
+        {
+            parameters.MultipleLocations = locations;
+
+            var result = await aggregator.GetBestEmissionsDataAsync(parameters);
+           
+            if(result != null)
+            {
+                var serializedOuput = JsonSerializer.Serialize((EmissionsDataDTO)result);
+                context.Console.WriteLine(serializedOuput);
+            }
+        }
+        else if (average) 
+        {
+            List<EmissionsDataDTO> emissions = new();
 
-        context.Console.WriteLine(JsonSerializer.Serialize(results));
+            foreach (var location in locations!)
+            {
+                parameters.SingleLocation = location;
+
+                var averageCarbonIntensity = await aggregator.CalculateAverageCarbonIntensityAsync(parameters);
+                
+                // If startTime or endTime were not provided, the aggregator would have thrwon an error. So, at this point it is safe to assume that the start/end values are not null. 
+                var emissionData = new EmissionsDataDTO()
+                {
+                    Location = location,
+                    Time = startTime,
+                    Duration = endTime - startTime,
+                    Rating = averageCarbonIntensity
+                };
+
+                emissions.Add(emissionData);
+            }
+            var serializedOuput = JsonSerializer.Serialize(emissions);
+            context.Console.WriteLine(serializedOuput);
+        }
+        else
+        {
+            parameters.MultipleLocations = locations;
+            var results = await aggregator.GetEmissionsDataAsync(parameters);
+            if (results != null)
```

Comment:
if the results are `null` should we still `exit 0`?  Or do we want to raise an exception `?? throws new SomeException()`?  Thoughts?

Reply:
Good point. I think it is a valid case if results are null, so we should not raise an exception. I changed the code so that it outputs an empty list which would indicate that no data was found for the requested parameters